File: sdk/unity/sdk/Assets/KingstonDynamics/DragonSong/Protocol/Messaging/Type/Heartbeat/HeartbeatPong.cs
using KingstonDynamics.DragonSong.Protocol.Messaging.Data;
using KingstonDynamics.DragonSong.Protocol.Utils;

namespace KingstonDynamics.DragonSong.Protocol.Messaging.Type.Heartbeat
{
    public class HeartbeatPong : Message
    {
        private const MessageType Type = MessageType.HEARTBEAT_PING;

        public HeartbeatPong(string auditId) : base((int)Type, auditId)
        {
            // Empty
        }

        public HeartbeatPong() : this(Keyinator.GenerateGUID())
        {
            // Empty
        }

        public HeartbeatPong(Readinator reader) : base(reader)
        {
            // Empty
        }
    }
}

File: src/protocol/src/main/java/com/kingstondynamics/dragonsong/protocol/message/type/heartbeat/HeartbeatPong.java
package com.kingstondynamics.dragonsong.protocol.message.type.heartbeat;

import com.kingstondynamics.dragonsong.protocol.message.Message;
import com.kingstondynamics.dragonsong.protocol.message.data.MessageType;
import com.kingstondynamics.dragonsong.protocol.utils.Keyinator;
import com.kingstondynamics.dragonsong.protocol.utils.Readinator;

public class HeartbeatPong extends Message {
    
    private final static MessageType type = MessageType.HEARTBEAT_PONG;
    
    public HeartbeatPong(String auditId) {
        super(type.getType(), auditId);
    }
    
    public HeartbeatPong() {
        this(Keyinator.generateGUID());
    }
    
    public HeartbeatPong(Readinator reader) {
        super(reader);
    }
}
